lib/Plack/Handler/FCGI.pm

```diff
@@ -162,6 +164,15 @@ sub run {
 
         $proc_manager && $proc_manager->pm_post_dispatch();
 
+        {
+            local $SIG{TERM} = sub {
```

Comment:
i'm not sure what you're trying to achieve with this. Maybe you're trying to capture stuck cleanup handlers? In that case, pushing another callback won't actually help, because the `for` loop only runs once?

Reply:
This fixes the case where the fcgi-manager exits and sends a TERM to the workers while the worker is busy processing the cleanup handlers. My testing shows in that case the worker wouldn't actually exit.  I didn't attempt to track down why, just that this fixed it.

Reply:
I mean, pushing to `psgix.cleanup.handlers` array has no effect because when a cleanup handler is running, the code is already inside the `for` loop for it.

Reply:
Yes, apparently I should not be [relying on an implementation detail](http://www.perlmonks.org/?node_id=1023497).  Oops.  My use-case has the manager, so this was a bit of an oversight. (although it does work in my testing)

Reply:
Oof, yeah, i'm surprised to see that it actually works, but yeah, i think it should be better not relying on it and use a regular variable and flags for this. I would also avoid using `$_` for no special reasons and use a regular `for my $handler ` loop.

Also can you add a source comment explaining this TERM logic that you described above? Thanks,